ipfs/provider.go

```diff
@@ -3,8 +3,40 @@ package ipfs
 import (
 	"io"
 
-	coreiface "github.com/ipfs/interface-go-ipfs-core"
+	blockstore "github.com/ipfs/go-ipfs-blockstore"
+	"github.com/ipfs/go-ipfs/core"
+	ipld "github.com/ipfs/go-ipld-format"
+	"github.com/libp2p/go-libp2p-core/routing"
 )
 
 // APIProvider allows customizable IPFS core APIs.
-type APIProvider func() (coreiface.APIDagService, io.Closer, error)
+type APIProvider interface {
+	DAG() ipld.DAGService
+	Blockstore() blockstore.Blockstore
+	Routing() routing.ContentRouting
+
+	io.Closer
+}
```

Comment:
this has the side benefit of making it easy to add and remove methods should that be called for in the future